### PR TITLE
JENKINS-46017 Restore Github Issue implementation 

### PR DESCRIPTION
**Pull request by i386**
# Description

Not sure what happened to this but it was stomped on earlier. Originally in https://github.com/jenkinsci/blueocean-plugin/pull/1300 and passed review.

See [JENKINS-46017](https://issues.jenkins-ci.org/browse/JENKINS-46017).

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by michaelneale** [comment]
as long as green get it in

**Comment by i386** [comment]
@vivek this seems to be failing with some dependency sadness. Could you PTAL?

**Comment by vivek** [comment]
@i386 is this rebased with latest master?

File: blueocean-github-pipeline/src/test/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssueTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+import java.util.Collection;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class GithubIssueTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
```
Comment (vivek) [inline blueocean-github-pipeline/src/test/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssueTest.java]: There is conflict of incompatible jetty libraries introduced in different versions of it and so wire mock and jenkins-test-harness do not play well by default. Basically we can't use default JenkinsRule. Replace this line with:

```
public JenkinsRule j = new PipelineBaseTest.PipelineBaseJenkinsRule();
```

Comment (i386) [reply blueocean-github-pipeline/src/test/java/io/jenkins/blueocean/blueocean_github_pipeline/GithubIssueTest.java]: NICE! Thanks @vivek 💯

**Pull request merged**